doc/write_ex.md

```diff
@@ -0,0 +1,147 @@
+# How to write an exercise?
```

Comment:
Use a more explicit filename: write_ex.md -> how-to-write-an-exercise-with-autogen.md